### DISCUSSION TITLE
[gatsby-starter-minimal-blog]

### DISCUSSION THREAD

**Discussion by flyingnobita** [LekoArts/gatsby-themes#993] [Questions]
I'm trying to install with:
```
gatsby new minimal-blog LekoArts/gatsby-starter-minimal-blog
```

But when running `gatsby develop`, I get the following warning and error:

Warning:
```
warn Plugin gatsby-plugin-theme-ui is not compatible with your gatsby version 4.23.1 - It requires gatsby@^2.0.0 || ^3.0.0
warn Warning: there are unknown plugin options for "gatsby-plugin-mdx", configured by node_modules/@lekoarts/gatsby-theme-minimal-blog-core: lessBabel
Please open an issue at https://ghub.io/gatsby-plugin-mdx if you believe this option is valid.
warn Plugin gatsby-plugin-theme-ui is not compatible with your gatsby version 4.23.1 - It requires gatsby@^2.0.0 || ^3.0.0
success load plugins - 7.095s
warn gatsby-plugin-react-helmet: Gatsby now has built-in support for modifying the document head. Learn more at https://gatsby.dev/gatsby-head
```

Error:
```
Module not found: Error: Can't resolve 'gatsby-plugin-mdx/loaders/mdx-loader' in '/workspaces/cv-website-gatsby-minimal-blog/site'
```

I tried installing `gatsby-plugin-mdx` and update my node modules with `npm update` but that didn't help.

I'm currently running Gatsby 4.18.2 (the default version). Does the above errors imply that I should downgrade my Gatsby to v3 instead or is there something else I'm missing?

Here's my package.json for reference:

```
{
  "private": true,
  "name": "minimal-blog",
  "description": "Typography driven, feature-rich blogging theme with minimal aesthetics. Includes tags/categories support and extensive features for code blocks such as live preview, line numbers, and line highlighting.",
  "version": "1.3.12",
  "author": "LekoArts <hello@lekoarts.de>",
  "license": "0BSD",
  "starter-name": "gatsby-starter-minimal-blog",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "develop:cypress": "cross-env CYPRESS_SUPPORT=y yarn develop",
    "build:cypress": "cross-env CYPRESS_SUPPORT=y yarn build",
    "start": "gatsby develop",
    "serve": "gatsby serve",
    "clean": "gatsby clean"
  },
  "dependencies": {
    "@lekoarts/gatsby-theme-minimal-blog": "^4.0.0",
    "gatsby": "^4.18.2",
    "gatsby-omni-font-loader": "^2.0.0",
    "gatsby-plugin-feed": "^4.18.1",
    "gatsby-plugin-gatsby-cloud": "^4.18.1",
    "gatsby-plugin-manifest": "^4.18.1",
    "gatsby-plugin-sitemap": "^5.18.1",
    "gatsby-plugin-webpack-bundle-analyser-v2": "^1.1.27",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  },
  "devDependencies": {
    "cross-env": "^7.0.3"
  }
}
```

Thank you.

**Comment by matheusdamiao** [comment]
I got exaclty the same error when trying to build a cloned version of `@lekoarts/gatsby-theme-emilia`. 
I also tried to install mdx plugin, because of the error, but it didn't work.

At last, I tried to downgrade my Gatsby from 4.17.1 to 3.00.0 in order to fix, but it turned out to get more errors than before, because lots of other plugins were not compatible with Gatsby versions < 4.00.0. 
So, I changed again, and set my Gatsby version in the package.json to###  4.00.0, reinstalled node modules, and then finally I could run develop successfully. 

So, @flyingnobita you can give it a try and see if it also works for you.

It's working for me, but I'm still getting a warning when in develop mode regarding theme-ui plugin. I don't know if that will bring any problem when editing/shadowing.

`warn Plugin gatsby-plugin-theme-ui is not compatible with your gatsby version 4.24.0 - It requires gatsby@^2.0.0 || ^3.0.0`

**Comment by flyingnobita** [comment]
That's very interesting.

What worked for me at the end was to follow the 'git clone' method.

**Comment by LekoArts** [comment]
You can ignore the `gatsby-plugin-theme-ui`, it's working fine.

I released a new major version with https://github.com/LekoArts/gatsby-themes/blob/main/themes/gatsby-theme-minimal-blog/CHANGELOG.md#500, so the problems mentioned are resolved and you can use the latest Theme UI and MDX versions.

**Reply by flyingnobita** [reply]
Thanks for the release!

I tried to install it and got the following warning:

![image](https://user-images.githubusercontent.com/46126470/193583758-bf1de4c6-cd3f-440d-8473-1b33df48028f.png)

Then for some reason (I have no idea if it's related to the above warnings) the code block isn't working:

![image](https://user-images.githubusercontent.com/46126470/193583906-bebba179-8480-4144-abd5-132b03c4219c.png)

I noticed the following in the Console of Chrome:

![image](https://user-images.githubusercontent.com/46126470/193584132-7925e538-adcb-4455-936d-0fc5dc73eb21.png)

**Reply by LekoArts** [reply]
Try a `git clone` of https://github.com/LekoArts/gatsby-starter-minimal-blog then and install with `npm install --legacy-peer-deps`. npm is doing something stupid again it seems
